src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs

```diff
@@ -867,16 +867,21 @@ private static void PopulateDownloadDependencies(IList<DownloadDependency> downl
 
                     if (!string.IsNullOrEmpty(versionValue))
                     {
-                        try
-                        {
-                            version = VersionRange.Parse(versionValue);
-                        }
-                        catch (Exception ex)
+                        var versions = versionValue.Split(';');
```

Comment:
Use `StringSplitOptions.RemoveEmptyEntries`.  As is, your changes expose a new exception code path.

I would use the same option [here](https://github.com/NuGet/NuGet.Client/pull/2929/commits/b258bd9009caa0fe4e1461efce4ae699c2794220#diff-bdb9fc4a876c0965afc8edfc0b807326R649) too, but it's not so terrible as-is since `GetVersionRange(…)` already performs a null check.

In general, I prefer filtering out garbage early.